src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ICollectionExtensions.cs

```diff
@@ -26,6 +26,18 @@ public static void AddRange<T>(this ICollection<T> collection, IEnumerable<T>? v
             }
         }
 
+        public static void AddRange<T>(this ICollection<T> collection, HashSet<T>? values)
```

Comment:
the caller is in SymbolUsageAnalysis doing this:

![image](https://user-images.githubusercontent.com/4564579/228906942-81f716b2-93f5-49c3-a1f8-f1aeea61308c.png)

Prior to this change, the AddRange call was calling into an IEnumerable taking method, causing the .GetEnumerator to returned a boxed enumerator.  With this change, it now calls this more specific overload which can iterate the hashset without allocating.